src/libfetchers/memory-input-accessor.cc

```diff
@@ -1,48 +1,43 @@
 #include "memory-input-accessor.hh"
+#include "memory-source-accessor.hh"
 
 namespace nix {
 
 struct MemoryInputAccessorImpl : MemoryInputAccessor
 {
-    std::map<CanonPath, std::string> files;
+    MemorySourceAccessor a;
```

Comment:
Can't we just make `MemoryInputAccessorImpl` a subclass of `MemorySourceAccessor` and `InputAccessor`? Then we don't need all these forwarding methods.

Reply:
@edolstra I we'll need to make the base `SourceAccessors` virtual but then yes we can.

Reply:
OK it's done!